### PR TITLE
Build time feature flags (Phase 1)

### PR DESCRIPTION
**Pull request by tstirrat**
Introduces build time feature flagging. Flags are build time, and cannot change at runtime yet (this will be added later)

- Uses https://github.com/garbles/flag for React feature flagging.
- Uses `react-native-config` to read the `.env` and parse `flag_*=true` vars into flag booleans
- Provides a `.env.release` which should allow configuration of flags for release only

#### Linked issues:

None

#### Screenshots:

None

#### How to test

* `yarn run-android` to see that Google import is available in settings
* Build an android release version to check that it is now disabled

### CODE REVIEW

File: app/views/Settings.js [hunk 1]
```diff
@@ -128,11 +128,14 @@ export const SettingsScreen = ({ navigation }) => {
...
         </Section>
 
-        {__DEV__ && (
-          <Section>
-            <GoogleMapsImport navigation={navigation} />
-          </Section>
-        )}
+        <Flag
+          name={['google_import']}
+          render={() => (
```
[the remark this answers is not in the document]
Comment (tstirrat) [reply app/views/Settings.js:133]: Ok, will look into this

File: .env [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Any variables that start with flag_ will be available to the <Flag> component
+# For now, only true/1 is parsed, everything else will be interpretted as false
+# flags must begin with flag_ and are case sensitive
+flag_google_import=true
```
[the remark this answers is not in the document]
Comment (tstirrat) [reply .env:4]: It's case sensitive, so it should remain as `flag_google_import`

**Comment by tstirrat** [comment]
Tested builds:

**Android:**

```
$ ./gradlew assembleRelease --stacktrace --no-daemon
...
> Configure project :app
Reading env from: .env.release <--------------------
```


Tested with manual overrides:

```bash
$ ENVFILE=.env.beta ./gradlew assembleRelease --stacktrace --no-daemon
...
> Configure project :app
Reading env from: .env.beta <---------------------
```

**iOS:**

```
$ yarn run-ios 
```

Screen Shot 2020-04-30 at 10 49 49 AM


```
$ ENVFILE=.env.release yarn run-ios
```

Screen Shot 2020-04-30 at 10 48 07 AM

File: app/views/Settings.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tstirrat) [reply app/views/Settings.js]:
```jsx
<Feature name="x" fallback={() => ...}>
  <FeatureUI />
</Feature>
```

**Comment by tstirrat** [comment]
@efalkner @kenpugsley Android is working with automatic debug/release config.

I don't know your iOS process but you should prepend `ENVFILE=.env.beta` to beta build commands and `ENVFILE=.env.release` to release build commands.

**Comment by efalkner** [comment]
Last I recall on the 🍎 build system, it was a tweak in XCode in the build settings, to tell it to interpret a different .env file. It always seemed to be flaky for me, and there be 🐉 🐉 . I'd recommend maybe copying the values over in CI, but that's just me ;)

**Comment by efalkner** [comment]
looks like I can make it work :D

**Comment by kenpugsley** [comment]
> looks like I can make it work :D

@efalkner - did you figure out how to do the release build on iOS?  Not working for me.

**Comment by tstirrat** [comment]
@kenpugsley  The underlying lib suggests this in CI before a release build:

```
echo ".env.release" > /tmp/envfile
```

Perhaps that's what you can do for now instead of `ENVFILE=.env.release`

File: .env [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# This is the debug/dev .env file. Any debug or non-release-signed build will
```
[the remark this answers is not in the document]
Comment (tstirrat) [reply .env:1]: Sort of..

iOS doesnt have a way to choose the base file without using `ENVFILE=`.. so the default debug mode would pick up nothing for iOS and `.env.dev` for Android.. so debug modes would look out of sync.

Thats mostly why I used .env for debug mode

[the remark this answers is not in the document]
Comment (Patrick-Erichsen) [reply .env:1]: Ah yeah, that makes sense. Probably makes to leave as is then.

**Comment by tstirrat** [comment]
This appears to be waiting on @efalkner to confirm if dev/release .env works in fastlanes flow

**Comment by tstirrat** [comment]
I just tested this with `ENVFILE=.env.release` where the google import code ONLY exists in `.env.release`:

Screen Shot 2020-05-03 at 1 17 43 PM


And then rebuilt with `ENVFILE=.env.beta` (has no flags):

Screen Shot 2020-05-03 at 1 20 25 PM


So it looks as though `ENVFILE=` works correctly for iOS releases. I updated the github release action to include `ENVFILE=.env.release` for any release builds made from there.

File: app/components/__tests__/Feature.spec.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import { render } from '@testing-library/react-native';
+import React from 'react';
+import { Text } from 'react-native';
+
+import { FlagsProvider } from '../../helpers/flags';
+import { Feature } from '../Feature';
+
+it('renders feature if the flag is enabled', () => {
```
Comment (efalkner) [inline app/components/__tests__/Feature.spec.js:8]: 😍

**Pull request merged**